modules/system/applications.nix

```diff
@@ -37,11 +38,40 @@ in
         rm ~/Applications/'Nix Apps'
       fi
 
-      if [ ! -e '/Applications/Nix Apps' ] \
-         || ourLink '/Applications/Nix Apps'; then
-        ln -sfn ${cfg.build.applications}/Applications '/Applications/Nix Apps'
+      ourFolder () {
+        local marker
+        marker=$(xattr -p org.nix-darwin "$1" 2>/dev/null)
+        [ "$marker" = 'generated' ]
+      }
+
+      targetFolder='/Applications/Nix Apps'
+
+      # Clean up old style link to nix store
+      if [ -e "$targetFolder" ] && ourLink "$targetFolder"; then
+        rm "$targetFolder"
+      fi
+
+      if [ -e "$targetFolder" ] && ! ourFolder "$targetFolder"; then
+        echo "warning: $targetFolder is not owned by nix-darwin, skipping App linking..." >&2
       else
-        echo "warning: /Applications/Nix Apps is not owned by nix-darwin, skipping App linking..." >&2
+        # create and mark folder
+        mkdir -p "$targetFolder"
+        xattr -w org.nix-darwin generated "$targetFolder"
+
+        rsyncFlags=(
+          --archive
+          --checksum
+          --chmod=-w
+          --copy-unsafe-links
+          --delete
+          --no-group
+          --no-owner
+        )
```

Comment:
What's the benefit of using an array for options here instead of simply adding the options one per line with a line continuation below in the actual `rsync` command?

Reply:
That originated from my module, so I’ll explain why I did it: I generally don’t like splitting commands over multiple lines. I’d rather declare an array with the flags and keep the command itself on one line. Arrays also work better with comments (e.g., should why it’s using `--copy-unsafe-links` be documented?).

Reply:
Thanks for the explanation, @reckenrode, much appreciated. Getting more context from comments on the used flags is very, very helpful; thanks @dwt on adding those 💯